### DISCUSSION TITLE
Page not found after refresh

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#14629] [Archive]
Hi,

I'm trying to understand why I'm landing on an "empty" interface page each time I refresh a page.
It happens for any admin pages ( settings, users or collections)

In case I start my navigation from `/admin` and I follow only links, i do not have any issues.

Note: It happens only on my production environnement.

I don't see any failing api call from the debug console that would show anything obvious. All the collection seems to be retrieved properly.

Is there anything special I could check to troubleshoot that issue ?

Thanks for help

![Page non trouvée - Google Chrome 25 07 2022 16_08_44](https://user-images.githubusercontent.com/1532438/180797934-65b4ed29-72e5-4eea-86d8-f4d5c6ff98ff.png)

**Comment by br41nslug** [comment]
Do you perhaps have a reverse proxy in front of your production environment that could be interfering with the App routing?

**Reply by gkielwasser** [reply]
Nope I don't have any.

**Reply by rijkvanzanten** [reply]
How are you hosting your production environment?

**Reply by gkielwasser** [reply]
I have an instance running on Google cloud run. I have also mapped a custom domain.

**Reply by rijkvanzanten** [reply]
OK so there's is a reverse proxy in front then to get the domain pointed to the Directus instance 🙂 

Is `/admin` in the root? Maybe the first load of the app to directus points to /something/else/admin 🤔

**Reply by gkielwasser** [reply]
Yes it is on the root of my subdomain

**Reply by gkielwasser** [reply]
Also to be more specific, I'm using this Mapping custom domains feature to link my domain to the one of cloud run.
https://cloud.google.com/run/docs/mapping-custom-domains?_ga=2.118824640.-877932701.1654270365#run

**Reply by rijkvanzanten** [reply]
No clue! I'm not familiar with cloud run or google's experimental custom domain mappings. The behavior you're seeing definitely makes me think there's something going wrong in the domain mapping somewhere somehow that makes the Vue app think it's on a different path. Did you set the `PUBLIC_URL` environment variable correctly?

**Reply by gkielwasser** [reply]
Yep it is.

From my desktop I can reproduce it 100% on Chrome and Edge ( I tried to double clean the browser cache).

From my mobile (Chrome browser), I cannot reproduce it, the page refresh well and the routing is applying correctly

**Reply by rijkvanzanten** [reply]
Huh, that doesn't really make any sense 🤔 Are you using the same user on all platforms? If it's a routing related problem, it should be consistently broken across every platform

**Reply by gkielwasser** [reply]
Yes I agree. I'm bit sceptic it is working on my mobile.

Yes i use in both cases an admin user. Also I refresh the exact same page.

**Reply by rijkvanzanten** [reply]
> Yes i use in both cases an admin user

To rule out any weirdness: the _same_ admin user, or another admin user in the same role? The app will store the last page the user was on for first login, so there's also a chance some logic goes wrong in there 🤔

**Reply by gkielwasser** [reply]
I'm logging each time with the same user which has an admin role.

**Reply by gkielwasser** [reply]
OK I have something more understandable! I logged out (with real logout) my user in my mobile. Then I loggedin again,

Now I have the same "page not found" issue

**Reply by rijkvanzanten** [reply]
Interesting! Every time you navigate in the app, the app attempts to update the current user's "last page" value in directus_users, and uses that page on app startup to let you pick up where you left off. It sounds like that value isn't correct (for some reason). (Although that being said, first reload on an existing path, like `/admin/settings/example`  shouldn't use that value 🤔 )

**Reply by gkielwasser** [reply]
From the `/me` api I can see well a correct value for the `last_page`

Also I tried to updated it with a known value `/admin/settings/project`.

The last page is well updated but the error persits.

**Reply by rijkvanzanten** [reply]
Hmm, so maybe a routing thing after all 🤔

**Reply by gkielwasser** [reply]
Would you see a reason loggin out my user would have changed how it behaves on page refresh ?

Something changed just after that logout/login 🤔.

I don't see why a routing issue could impact that.

**Reply by rijkvanzanten** [reply]
Are you seeing the page not found on first login from `/admin`, or on a "hard refresh" on a path like `/admin/content/articles` ?

**Reply by gkielwasser** [reply]
The first routing after login works fine and I land and the correct page.

It is the hard refresh that is failing.

Additionnal note: I got it working back on my mobile. The only operation I have done is logout/login from my mobile.

From the desktop it is perfectly "constant", it doesn't work :D

**Reply by gkielwasser** [reply]
OK, I have something new! There is for sure a timing issue or stuff like that. I can have it working or not just doing some concurrent refresh from my mobile.

Maybe I cannot reproduce this situation on my desktop.

**Comment by dimmkan** [comment]
Good afternoon. I am facing a similar problem in my Directus instance. As mentioned above, there may be problems with routing. I have Nginx as a reverse proxy in front of my application, but I have absolutely no idea how to fix this situation? Why when clicking on a specific link, for example https://cms.promoydom.ru/admin/content/news , the application outputs "Page not found". At the same time, when you click from the start page, the links are clicked.

**Reply by dimmkan** [reply]
Up this theme)) I really hope for your help)

**Reply by rijkvanzanten** [reply]
Sounds like a nginx configuration issue, which is out of our control.. Avoiding this sort of problem is why we started Directus Cloud ^^

**Reply by dimmkan** [reply]
Thank you very much for your reply) I do not know how it worked, but after your answer I tried again to analyze the Nginx settings and found what the problem was))

**Comment by dimmkan** [comment]
![image](https://user-images.githubusercontent.com/18102760/199371026-51759568-9661-4840-9e5b-fa42ea0493a6.png)

**Reply by dimmkan** [reply]
The problem lay in a small slash, which was missing at the end of the reverse proxy address))

**Comment by dimmkan** [comment]
And yet there is no (( Perhaps it was a coincidence, but when re-checking the error began to repeat again. :(
